### PR TITLE
Wishlist Page

### PR DESCRIPTION
**Pull request by praveenreddy1798**
wishlist

### CODE REVIEW

File: src/containers/Wishlist.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          {!loading && (
+            <h1 class="flex-center">
+              {wishlist.length ? "Wishlist" : "Your Wishlist is empty"}
+            </h1>
+          )}
+          <Loader loading={loading} />
+          <div class="grid responsive-grid pd-sm">
+            {wishlist.map((product) => {
+              const { _id } = product;
+              return <Card key={_id} product={product} cardType="wishlist" />;
```
Comment (vaibhav18matere) [inline src/containers/Wishlist.js]: change `class`  to `className`

File: src/services/Auth/login.js [hunk 2]
```diff
@@ -3,38 +3,36 @@ import { useNavigate } from "react-router-dom";
 import { useProducts } from "../../context/products";
 import { useToast } from "../../context/toast";
 import { useAxios } from "../../hooks";
-import { MESSAGES } from "../../utils/constants";
+import { MESSAGES } from "../../utils";
```
Comment (vaibhav18matere) [inline src/services/Auth/login.js:7]: Is there any specific reason of importing `MESSAGE` in caps?

File: src/App.js [hunk 3]
```diff
@@ -4,16 +4,19 @@ import ProductsListing from "./containers/ProductsListing";
 import Home from "./containers/Home";
 import Login from "./containers/Login";
 import Signup from "./containers/Signup";
+import Wishlist from "./containers/Wishlist";
```
Comment (junaidshaikh-js) [inline src/App.js]: Read about barrel imports, it helps to write clean code.

File: src/services/wishlist.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  }, [data?.wishlist, errorMessage, productsDispatch, toastDispatch]);
+  return { loading, addToWishlist };
+};
+
+export const useRemoveFromWishlist = () => {
+  const [url, setUrl] = useState(null);
+  const [enabled, setEnabled] = useState(false);
+  const [product, setProduct] = useState(null);
+  const { productsDispatch } = useProducts();
+  const { toastDispatch } = useToast();
```
Comment (vaibhav18matere) [inline src/services/wishlist.js:74]: I think all the exports should be on top and logic below it. 
Please check and confirm

**Review by vaibhav18matere — APPROVED** [review]

File: src/containers/Home.js [hunk 5]
```diff
@@ -8,7 +7,6 @@ const Home = () => {
     <div class="page-wrapper page-wrapper-filter-mob">
       <Navbar />
       <main class="main-section main-section-strech pd-md">
```
Comment (junaidshaikh-js) [inline src/containers/Home.js]: In JSX, we need to write className instead of class

File: src/containers/Wishlist.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { Card } from "../components/Card";
+import { Loader } from "../components/Loader";
+import { Navbar } from "../components/Navbar";
+import { useProducts } from "../context/products";
+import { useQueryWishlistProducts } from "../services";
+const Wishlist = () => {
+  const { productsState } = useProducts();
+  const { wishlist } = productsState;
```
Comment (junaidshaikh-js) [inline src/containers/Wishlist.js]: A wishlist can be directly destructured 

```
const { productsState: { wishlist }} = useProducts();
```

**Review by junaidshaikh-js — COMMENTED** [review]
Some changes are suggested. 

- You can use barrel import
- You need to change the class with className 

Everything else looks good.

File: src/services/Auth/login.js (hunk 2 above)
Comment (praveenreddy1798) [reply src/services/Auth/login.js:7]: Its a constant. Constants are usually in caps

Comment (vaibhav18matere) [reply src/services/Auth/login.js:7]: > Its a constant. Constants are usually in caps

got it!

**Comment by praveenreddy1798** [comment]
Made all the review changes. Merging them

**Pull request merged**